app/src/main/java/com/example/unscramble/ui/GameScreen.kt

```diff
@@ -0,0 +1,226 @@
+            .size(48.dp),
+    ) {
+        Text(
+            text = stringResource(R.string.word_count, wordCount),
+            fontSize = 18.sp,
+        )
+        Text(
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentWidth(Alignment.End),
+            text = stringResource(R.string.score, score),
+            fontSize = 18.sp,
+        )
+    }
+}
+
+@Composable
+fun GameLayout(
+    currentScrambledWord: String,
+    isGuessWrong: Boolean,
+    userGuess: String,
+    onUserGuessChanged: (String) -> Unit,
+    onKeyboardDone: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        verticalArrangement = Arrangement.spacedBy(24.dp)
+    ) {
+        Text(
+            text = currentScrambledWord,
+            fontSize = 45.sp,
+            modifier = modifier.align(Alignment.CenterHorizontally)
+        )
+        Text(
+            text = stringResource(R.string.instructions),
+            fontSize = 17.sp,
+            modifier = Modifier.align(Alignment.CenterHorizontally)
+        )
+        OutlinedTextField(
+            value = userGuess,
+            singleLine = true,
+            modifier = Modifier.fillMaxWidth(),
+            onValueChange = onUserGuessChanged,
+            label = {
+                if (isGuessWrong) {
+                    Text(stringResource(R.string.wrong_guess))
+                } else {
+                    Text(stringResource(R.string.enter_your_word))
+                }
+            },
+            isError = isGuessWrong,
+            keyboardOptions = KeyboardOptions.Default.copy(
+                imeAction = ImeAction.Done
+            ),
+            keyboardActions = KeyboardActions(
+                onDone = { onKeyboardDone() }
+            ),
+        )
+    }
+}
+
+/*
+ * Creates and shows an AlertDialog with final score.
+ */
+@Composable
+private fun FinalScoreDialog(
+    score: Int,
+    onPlayAgain: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val activity = (LocalContext.current as Activity)
+
+    AlertDialog(
+        onDismissRequest = {
+            // Dismiss the dialog when the user clicks outside the dialog or on the back
+            // button. If you want to disable that functionality, simply use an empty
+            // onCloseRequest.
+        },
+        title = { Text(stringResource(R.string.congratulations)) },
+        text = { Text(stringResource(R.string.you_scored, score)) },
+        modifier = modifier,
+        dismissButton = {
+            TextButton(
+                onClick = {
+                    activity.finish()
+                }
+            ) {
+                Text(text = stringResource(R.string.exit))
+            }
+        },
+        confirmButton = {
+            TextButton(onClick = onPlayAgain) {
+                Text(text = stringResource(R.string.play_again))
+            }
+        }
+    )
+}
+
+@Preview(showBackground = true)
+@Composable
+fun GameScreenPreview() {
```

Comment:
should this be in the MainActivity file? so we don't need to import Unscramble theme here?

Reply:
Answered above